### PR TITLE
Adds a damage threshold + RNG requirement for EMPs affecting heads and torsos, as well as organs inside those two.

### PR DESCRIPTION
**Pull request by BurgerLUA**
## About The Pull Request

Adds a damage threshold + RNG requirement for EMPs affecting heads and torsos, as well as organs inside those two.
For every point of damage on the bodypart (or if you're an organ, the bodypart you belong to), there is a +2% chance for that bodypart (or organ) to trigger the effects of EMP. This chance is doubled if it's a heavy EMP.

Example, if your torso has 25 damage, and you were affected by an EMP, your torso has a 50% chance to suffer the default effects of EMP. All your organs inside your torso will EACH roll the 50% chance to be affected by the EMP, regardless if your torso was EMP'd or not.

Graph here: https://www.desmos.com/calculator/7sdtn22uda

**Note that some effects can still trigger, such as stun and blindness, regardless of damage. This is intentional.**

## Why It's Good For The Game

EMPs are super strong against synths. Perhaps unreasonably strong. This PR nerfs this by making it so that you actually have to do some damage to the synth before you can EMP them for an instant win. The justification for this is below:

image

TL;DR, if your electronics are enclosed in steel, it won't suffer the effects of EMP unless there are holes (damage).

## Proof Of Testing

Draft because untested.

### Synth main????
I have never played synth because I am not sexually attracted to metal.

### Does this affect cyborgs?
This PR does not change cyborg EMP behavior.

### Does this affect legs and arms?
This PR does not change leg and arm EMP behavior, since they're jointed.

## Changelog

:cl: BurgerBB
balance: Adds a damage threshold + RNG requirement for EMPs affecting heads and torsos, as well as organs inside those two.
/:cl:

### CODE REVIEW

File: modular_zubbers/code/modules/emp_overhaul/organ.dm [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// https://www.desmos.com/calculator/7sdtn22uda
+
+/obj/item/bodypart/emp_act(severity)
+
+	if(!(src.biological_state & BIO_STANDARD_JOINTED) && !prob((src.brute_dam + src.burn_dam)*2*(2/severity))) //Chance to get EMP'd is based on self bodypart damage.
```
Comment (Majkl-J) [inline modular_zubbers/code/modules/emp_overhaul/organ.dm:5]:
```suggestion
	if(!(src.biological_state & BIO_STANDARD_JOINTED) && !prob((src.get_damage())*2*(2/severity))) //Chance to get EMP'd is based on self bodypart damage.
```

**Review by Majkl-J — CHANGES_REQUESTED** [review]

File: modular_zubbers/code/modules/emp_overhaul/organ.dm [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+		return EMP_PROTECT_ALL
```
[the remark this answers is not in the document]
Comment (BurgerLUA) [reply modular_zubbers/code/modules/emp_overhaul/organ.dm:6]: In a lot of act_emp calls, EMP_PROTECT_CONTENTS is returned. EMP_PROTECT_ALL contains that bitflag and more. I'm just doing the same thing.

[the remark this answers is not in the document]
Comment (Majkl-J) [reply modular_zubbers/code/modules/emp_overhaul/organ.dm:6]: `code/modules/surgery/bodyparts/_bodyparts.dm#L1409-L1412` Seems to be using `return FALSE`, we probably should keep consistent with the parent proc

**Review by Majkl-J — APPROVED** [review]
Looks good to me, once you're done testing it, undraft it and I'll TM it for a week just incase before merging.

**Review by projectkepler-ru — APPROVED** [review]

**Review by Majkl-J — CHANGES_REQUESTED** [review]
Due to how `code\modules\surgery\bodyparts\robot_bodyparts.dm` handles these, you need to return FALSE or refactor this upstream. Damage isn't applied but negative effects still are

**Comment by BurgerLUA** [comment]
> Due to how `code\modules\surgery\bodyparts\robot_bodyparts.dm` handles these, you need to return FALSE or refactor this upstream. Damage isn't applied but negative effects still are

Check the first file changed. This already was resolved.

**Comment by Majkl-J** [comment]
> > Due to how `code\modules\surgery\bodyparts\robot_bodyparts.dm` handles these, you need to return FALSE or refactor this upstream. Damage isn't applied but negative effects still are
> 
> Check the first file changed. This already was resolved.

Nope, robot parts still have old handling code. I tested your PR locally and still got a good half of the negative effects, like the ugly green screen

**Comment by BurgerLUA** [comment]
pain

**Comment by BurgerLUA** [comment]
This is ready to be merged.

**Review by Majkl-J — APPROVED** [review]
This will need some adjustments once I go and make organ and limb emp processing not happen independently of eachother

Looks fine for now though, love the work
